crypto/evp/pmeth_fn.c

```diff
@@ -20,6 +20,9 @@ static EVP_SIGNATURE *evp_signature_new(OSSL_PROVIDER *prov)
 {
     EVP_SIGNATURE *signature = OPENSSL_zalloc(sizeof(EVP_SIGNATURE));
 
+    if (signature == NULL)
+        return NULL;
+
```

Comment:
You could cascade this with the lock creation check and have them both set `ERR_R_MALLOC_FAILURE`